src/Agent.Sdk/Knob/Knob.cs

```diff
@@ -24,14 +24,25 @@ public ExperimentalKnob(string name, string description, params IKnobSource[] so
         }
     }
 
+    public class SecretKnob : Knob
+    {
+        public override bool isSecret => true;
```

Comment:
Why do we need the flag if there is a SecretKnob type already? Can we use condition like 'object is SecretKnob' to determine it?